### PR TITLE
Allow overlapping elution profiles in elution peak detection

### PR DESCRIPTION
**Pull request by timosachsenberg**
# Description

With a large enough window for maxima detection, the "neighbor peaks" overlap on the test data:
image
Before the patch, only one maximum (the left one) was found because the algorithm breaks at the first overlapping peak.
With the patch, the collection of peaks can go on and both maxima are found.

# Checklist:
- [x] Make sure that you are listed in the AUTHORS file
- [x] Add relevant changes and new features to the CHANGELOG file
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes
- [x] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.
- Or click on the action: e.g., for clang-format linting

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

# Advanced commands (admins / reviewer only):
- /rebase will try to rebase the PR on the current develop branch.
- /reformat (experimental) applies the clang-format style changes as additional commit
- setting the label NoJenkins will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)

### CODE REVIEW

**Comment by timosachsenberg** [comment]
With a large enough window for maxima detection, the neighbor peaks overlap on the test data:
image

**Comment by timosachsenberg** [comment]
number of mass traces changed from 448 to 449

File: src/tests/class_tests/openms/source/ElutionPeakDetection_test.cpp [hunk 1]
```diff
@@ -174,6 +174,23 @@ START_SECTION((void findLocalExtrema(const MassTrace &, const Size &, std::vecto
     // SavitzkyGolay
     TEST_EQUAL(maxes.size(), 4);
     TEST_EQUAL(mins.size(), 2);
+
+    std::cout << "maxima: " << maxes[0] << ", " << maxes[1] << ", "
```
Comment (JeeH-K) [inline src/tests/class_tests/openms/source/ElutionPeakDetection_test.cpp:178]: this test writing part should be deleted for the merge?

**Review by JeeH-K — COMMENTED** [review]
nothing seems to break with the new EPD attached to FLASHDeconvQ with my data.

File: src/tests/class_tests/openms/source/ElutionPeakDetection_test.cpp (hunk 1 above)
Comment (timosachsenberg) [reply src/tests/class_tests/openms/source/ElutionPeakDetection_test.cpp:178]: I can delete it

**Review by JeeH-K — APPROVED** [review]

**Pull request merged**